tgui/packages/tgui/components/Button.js

```diff
@@ -94,22 +95,33 @@ export const Button = props => {
         }
       }}
       {...computeBoxProps(rest)}>
-      {(icon && iconPosition !== 'right') && (
-        <Icon
-          name={icon}
-          color={iconColor}
-          rotation={iconRotation}
-          spin={iconSpin} />
-      )}
-      {content}
-      {children}
-      {(icon && iconPosition === 'right') && (
-        <Icon
-          name={icon}
-          color={iconColor}
-          rotation={iconRotation}
-          spin={iconSpin} />
-      )}
+      <div
+        className={
+          verticalAlign && typeof verticalAlign === 'string'
+            ? 'Button--align--' + verticalAlign
+            : undefined
+        }>
+        <div>
+          {icon && iconPosition !== 'right' && (
```

Comment:
First, I don't think it was necessary to add 2 divs, I think 1 would've worked well enough, one that wraps the text, since the icon is already a flex element.

What you can do, is add this to the base button class:

```
display: inline-block;
display: inline-flex;
```

This will display a button as an inline block on platforms where flex isn't supported, and will use an inline flex on IE11.

Then you need this in `Button--fluid`:

```
display: block;
display: flex;
```

Reply:
I implemented everything here but I couldn't get it to center properly with the div on the content.

<details closed>
<summary>Current Master</summary>
<img src="https://user-images.githubusercontent.com/54709710/140052379-fc60cc30-19aa-483f-a52f-a01e6c7e68a6.png">
</details>

<details closed>
<summary>Div Inside</summary>
<img src="https://user-images.githubusercontent.com/54709710/140051930-9597dc38-e8ec-4993-a426-dcff5815d6c3.png"> <br/>
<img src="https://user-images.githubusercontent.com/54709710/140051718-175ca745-eeda-4018-90d8-6ede42efab08.png"> <br/>
Which also seems to break normal width buttons. <br/>
<img src="https://user-images.githubusercontent.com/54709710/140051814-076e9962-581d-45bd-ab46-5e295a61394d.png"> <br/>
<img src="https://user-images.githubusercontent.com/54709710/140052109-4d49d3b0-923b-40d7-97d4-3b162c724041.png"> <br/>

</details>

<details closed>
<summary>Div Oustide (currently implemented)</summary>
<img src="https://user-images.githubusercontent.com/54709710/140051447-8b0c6a60-8367-4d98-8ecd-2eefa7a23cea.png"> <br/>
<img src="https://user-images.githubusercontent.com/54709710/140051500-d2ca6c5e-40db-4258-9a4e-beb431c8a8ff.png">
</details>